pkg/validation-webhook/webhook.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/spf13/cobra"
+
+	v1 "k8s.io/api/admission/v1"
+	"k8s.io/api/admission/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	// TODO: try this library to see if it generates correct json patch
+	// https://github.com/mattbaird/jsonpatch
+)
+
+var (
+	certFile string
+	keyFile  string
+	port     int
+)
+
+// CmdWebhook is used by agnhost Cobra.
+var CmdWebhook = &cobra.Command{
+	Use:   "validation-webhook",
+	Short: "Starts a HTTP server, useful for testing MutatingAdmissionWebhook and ValidatingAdmissionWebhook",
+	Long: `Starts a HTTP server, useful for testing MutatingAdmissionWebhook and ValidatingAdmissionWebhook.
+After deploying it to Kubernetes cluster, the Administrator needs to create a ValidatingWebhookConfiguration
+in the Kubernetes cluster to register remote webhook admission controllers.`,
+	Args: cobra.MaximumNArgs(0),
+	Run:  main,
+}
+
+func init() {
+	CmdWebhook.Flags().StringVar(&certFile, "tls-cert-file", "",
```

Comment:
Can we document all these args in the README, similar to what we do for the other components?

Reply:
Added the webhook flags, left out the klog flags.